src/Symfony/Component/Security/Http/Firewall/AccessListener.php

```diff
@@ -68,4 +77,19 @@ public function handle(GetResponseEvent $event)
             throw new AccessDeniedException();
         }
     }
+
+    private function handleWithAuthorizationChecker(GetResponseEvent $event)
+    {
+        $request = $event->getRequest();
+
+        list($attributes) = $this->map->getPatterns($request);
+
+        if (null === $attributes) {
+            return;
+        }
```

Comment:
All the previous lines in the new methods duplicate the logic in "handle".
Would it make sense to not add any newmethod, and instead move the "if" block throwing "AuthenticationCredentialsNotFoundException" after these lines in "handle"?